### PR TITLE
[DCOS-40654] When listing reserved resources, fail test if we leaked some.

### PR DESCRIPTION
**Pull request by porridge**
This should let sdk_diag get the log for uninstall scheduler to help us
establish the cause for those leaks.

This is already implemented on master in a different way.

### CODE REVIEW

File: testing/sdk_utils.py [hunk 1]
```diff
@@ -25,18 +25,30 @@ def get_service_name(default: str) -> str:
...
-    state_json_slaveinfo = dcos.mesos.DCOSClient().get_state_summary()['slaves']
+def list_reserved_resources(service_name: str) -> bool:
+    """
+    Logs all reserved resources and returns a true value if there are some for this service's role.
 
-    for slave in state_json_slaveinfo:
+    The reservations are retrieved from state.json.
+    This is for debugging DCOS-40654.
+    """
+    service_role = service_name.strip("/").replace("/", "__") + "-role"
```
Comment (elezar) [inline testing/sdk_utils.py]: I think there is a `get_role()` function in `sdk_utils` now.

File: testing/sdk_utils.py [hunk 2]
```diff
@@ -25,18 +25,30 @@ def get_service_name(default: str) -> str:
...
-        if reserved_resources == {}:
+        if not reserved_resources:
             continue
-        msg = 'on slaveid=%s hostname=%s reserved resources: %s'
-        log.info(msg % (slave['id'], slave['hostname'], reserved_resources))
+        reserved_resources_messages.append(msg % (slave['id'], slave['hostname'], reserved_resources))
+        if service_role in reserved_resources:
+            found_for_this_service = True
+    if reserved_resources_messages:
+        log.info('Found following reserved resources post uninstall.')
```
Comment (elezar) [inline testing/sdk_utils.py]: Would it be useful to print the name of the service we are uninstalling?

Nit: This function doesn't really have anything with uninstalling.

File: testing/sdk_utils.py [hunk 3]
```diff
@@ -25,18 +25,30 @@ def get_service_name(default: str) -> str:
     return os.environ.get("INTEGRATION_TEST__SERVICE_NAME") or default
 
 
-def list_reserved_resources():
-    '''Displays the currently reserved resources on all agents via state.json;
-       Currently for INFINITY-1881 where we believe uninstall may not be
-       always doing its job correctly.'''
-    state_json_slaveinfo = dcos.mesos.DCOSClient().get_state_summary()['slaves']
+def list_reserved_resources(service_name: str) -> bool:
```
Comment (elezar) [inline testing/sdk_utils.py]: If we only call this on uninstall and with the service name, we could change the name of the function.

File: testing/sdk_utils.py (hunk 1 above)
Comment (elezar) [reply testing/sdk_utils.py]: (scratch that ... that was in `master`)

**Comment by porridge** [comment]
Yes, this is to make tests fail in the guilty test, and therefore gather useful logs.
Changes pushed, PTAL.

**Comment by porridge** [comment]
FTR I've cherry-picked your CI fixes into a separate PR:
https://github.com/mesosphere/dcos-commons/pull/2644

Once that merges, I will rebase this one, to keep the changes separate.

**Comment by porridge** [comment]
Rebased to see if we can get a successful run.

**Comment by porridge** [comment]
I'm holding off on merging this until we fix the root cause, since it would probably just make all of our CI red, given how frequently the new code finds a leak.

**Comment by kaiwalyajoshi** [comment]
Closing this PR as we're really trying to tombstone all frameworks on SDK 0.40.0 branches.

**Pull request closed without merging**